DataFormats/L1THGCal/src/HGCalCluster.cc

```diff
@@ -32,65 +35,42 @@ HGCalCluster::~HGCalCluster()
 }
 
 
-void HGCalCluster::addTriggerCell(const l1t::HGCalTriggerCell &tc)
+void HGCalCluster::addTriggerCell( const edm::Ptr<l1t::HGCalTriggerCell > &tc )
 {
-
+    
     if( triggercells_.size() == 0 ){ 
-        seedDetId_ = tc.detId();
-        seedMipPt_ = tc.mipPt();
+        seedDetId_ = tc->detId();
+        seedMipPt_ = tc->mipPt();
     }
 
     /* update cluster positions */
-    Basic3DVector<float> tcVector( tc.position().x(),
-                                   tc.position().y(),
-                                   tc.position().z() );
-
-    Basic3DVector<float> centreVector( tc.position().x(), 
-                                       tc.position().y(), 
-                                       tc.position().z() );
-
-    centreVector = centreVector*mipPt_ + tcVector*tc.mipPt();
-    centreVector = centreVector / ( mipPt_ + tc.mipPt() ) ;
-
-    GlobalPoint centreTmp((float)centreVector.x(), 
-                          (float)centreVector.y(), 
-                          (float)centreVector.z() );
+    Basic3DVector<float> tcVector( tc->position() );
+    Basic3DVector<float> centreVector( centre_ );
 
-    GlobalPoint centreProjTmp((float)centreVector.x() / centreVector.z(),  
-                              (float)centreVector.y() / centreVector.z(), 
-                              (float)centreVector.z() / centreVector.z() );
+    centreVector = centreVector*mipPt_ + tcVector*tc->mipPt();
+    centreVector = centreVector / ( mipPt_ + tc->mipPt() ) ;
 
-    centre_ = centreTmp;
-    centreProj_ = centreProjTmp;
+    centre_ = GlobalPoint( centreVector );
+    centreProj_= GlobalPoint( centreVector / centreVector.z() );
 
     /* update cluster energies */
-    mipPt_ += tc.mipPt();
+    mipPt_ += tc->mipPt();
 
-    int hwPt = 0;
-    hwPt  += tc.hwPt();
+    int hwPt = this->hwPt() + tc->hwPt();
     this->setHwPt(hwPt);
 
     math::PtEtaPhiMLorentzVector p4( ( this->p4() )  );
-    p4 += tc.p4(); 
+    p4 += tc->p4(); 
     this->setP4( p4 );
 
-}
-void HGCalCluster::addTriggerCellList( edm::Ptr<l1t::HGCalTriggerCell> &p)
-{
-
-    triggercells_.push_back( p );
-
+    triggercells_.push_back( tc );
 }
 
 
-double HGCalCluster::distance(const l1t::HGCalTriggerCell &tc) const
+double HGCalCluster::distance(const edm::Ptr<l1t::HGCalTriggerCell> tc) const
```

Comment:
It was fine before. An edm::Ptr is not needed here, so it is better to have more standard function arguments